src/alertmanager.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Alertmanager workload manager."""
+
+import logging
+import re
+from typing import List, Optional, Tuple
+
+from alertmanager_client import Alertmanager, AlertmanagerBadResponse
+from ops.pebble import ExecError, Layer
+
+from src.workload_manager import ConfigUpdateFailure, ContainerNotReady, WorkloadManager
+
+logger = logging.getLogger(__name__)
+
+
+class AlertmanagerWorkloadManager(WorkloadManager):
+    """Workload manager for Alertmanager."""
+
+    _exe_name = "alertmanager"
+
+    def __init__(
+        self,
+        charm,
+        *,
+        container_name: str,
+        peer_addresses: List[str],
+        api_port: int,
+        ha_port: int,
+        web_route_prefix: str,
+        external_url: str,
+        config_path: str,
+        web_config_path: str,
+        tls_enabled: bool,
+    ):
+        # Must inherit from ops 'Object to be able to register events.'
+        super().__init__(
+            charm=charm,
+            container_name=container_name,
+            external_url=external_url,
+            config_path=config_path,
+            tls_enabled=tls_enabled,
+        )
+
+        self._peer_addresses = peer_addresses
+        self._api_port = api_port
+        self._ha_port = ha_port
+        self.api = Alertmanager(port=self._api_port, web_route_prefix=web_route_prefix)
```

Comment:
proposal: rename 
Alertmanager -> Api
AlertmanagerWorkloadManager -> Workload
the module name shouldn't be repeated in all classes defined in it.